packages/cells/src/widget.ts

```diff
@@ -1577,7 +1577,7 @@ export class MarkdownCell extends AttachmentsCell<IMarkdownCellModel> {
       let collapseButton = this.inputArea.promptNode.appendChild(
         document.createElement('button')
       );
-      collapseButton.className = `jp-Button jp-mod-minimal ${HEADING_COLLAPSER_CLASS}`;
+      collapseButton.className = `jp-Button jp-mod-minimal ${HEADING_COLLAPSER_CLASS} ${HEADING_COLLAPSER_CLASS}-h${this.headingInfo.level}`;
       collapseButton.style.background = `${
         this._headingCollapsed
           ? 'var(--jp-icon-caret-right)'
```

Comment:
Would you mind changing the logic by not setting the `collapseButton.style.background` but rather by adding a class `jp-mod-collapsed` if `this._headingCollapsed` is true. This will allow to bring more flexibility on the CSS style (and remove the need of `!important`.

This means that by default the collapse button should have `background: var(--jp-icon-caret-down) no-repeat; background-position-x: center;` and when collapse `background-image: var(--jp-icon-caret-right)`.